File: CSharp/spotifyQueue.cs
using System;
using System.Collections.Generic;
namespace CSharp
{
    class spotifyQueue
    {
        static void Main(string[] args)
        {
            Queue<string> songQueue = new Queue<string>();
            int choice = 0;
            while (choice != -1)
            {
                Console.WriteLine("Welcome to the Spotify Queue System");
                Console.WriteLine("1) Queue Song");
                Console.WriteLine("2) Dequeue Song");
                Console.WriteLine("3) See Next Track");
                Console.WriteLine("4) Latest Added Song");
                Console.WriteLine("5) See Queue");
                Console.WriteLine("6) Clear Queue");
                Console.WriteLine("7) Exit");
                choice = Convert.ToInt32(Console.ReadLine());

                switch (choice)
                {
                    case 1:
                        Console.WriteLine("Enter Song Name");
                        string song = Console.ReadLine();
                        foreach (string s in songQueue)
                        {
                            if (s == song)
                            {
                                Console.WriteLine("Song Already in Queue");
                                song = null;
                                clearScreen();
                                break;
                            }
                        }
                        songQueue.Enqueue(song);
                        Console.WriteLine("Song Queued");
                        clearScreen();
                        break;
                    case 2:
                        Console.WriteLine("Queue: ");
                        foreach (string s in songQueue)
                        {
                            Console.WriteLine(s);
                        }
                        Console.WriteLine("What song do you want to remove?");
                        string removeSong = Console.ReadLine();
                        if (songQueue.Contains(removeSong))
                        {
                            songQueue = new Queue<string>(songQueue.ToArray().Where(val => val != removeSong));
                            Console.WriteLine("Song Removed");
                            clearScreen();
                            break;
                        }
                        Console.WriteLine("Song not in Queue");
                        clearScreen();
                        break;
                    case 3:
                        Console.WriteLine("Next Song is: " + songQueue.Peek());
                        clearScreen();
                        break;
                    case 4:
                        Console.WriteLine("Latest Song is: " + songQueue.ToArray()[songQueue.Count - 1]);
                        clearScreen();
                        break;
                    case 5:
                        Console.WriteLine("Queue is: ");
                        foreach (string s in songQueue)
                        {
                            Console.WriteLine(s);
                        }
                        clearScreen();
                        break;
                    case 6:
                        songQueue.Clear();
                        Console.WriteLine("Queue Cleared");
                        clearScreen();
                        break;
                    case 7:
                        choice = -1;
                        Console.Clear();
                        break;
                    default:
                        Console.WriteLine("Invalid Choice");
                        break;
                }
            }
            Console.WriteLine("Thank You for using Spotify Queue System");
        }

        static void clearScreen()
        {
            Console.WriteLine("Press any key to continue");
            Console.ReadKey();
            Console.Clear();
        }
    }
}

File: IB/spotifyQueue.java
import java.util.*;

public class spotifyQueue {
    public static void main(String[] args) {
        Queue<String> songQueue = new LinkedList<String>();
        int choice = 0;
        Scanner sc = new Scanner(System.in);
        while(choice != -1){
            System.out.println("Welcome to the Spotify Queue System\n" +
                            "1) Add a song to the queue\n" +
                            "2) Remove a song from the queue\n" +
                            "3) See Next Track\n" +
                            "4) See Latest Track\n" +
                            "5) View the queue\n" +
                            "6) Clear the queue\n" +
                            "7) Exit");
            String song, lastsong = "";
            Boolean found = false;
            choice = sc.nextInt();
            switch(choice){
                case 1:
                    System.out.println("Enter the song name: ");
                    song = sc.next();

                    for (String s : songQueue) {
                        if (s.equals(song)) {
                            System.out.println("Song already in queue");
                            found = true;
                            clearScreen();
                            break;
                        }
                    }

                    if (!found) {
                        songQueue.add(song);
                        lastsong = song;
                        System.out.println("Song added to queue");
                        clearScreen();
                    } else {
                        break;
                    }
                case 2:
                    System.out.println("Queue: ");
                    for (String s : songQueue) {
                        System.out.println(s);
                    }

                    System.out.println("What song would you like to remove?");
                    song = sc.next();

                    for(String s : songQueue){
                        if (s.equals(song)) {
                            songQueue.remove(song);
                            System.out.println("Song removed from queue");
                            found = true;
                            clearScreen();
                            break;
                        }
                    }
                    
                    if (!found) {
                        System.out.println("Song not found in queue");
                        clearScreen();
                        break;
                    } else {
                        break;
                    }
                case 3:
                    System.out.println("Next Track: " + songQueue.peek());
                    clearScreen();
                    break;
                case 4:
                    System.out.println("Latest Track: " + lastsong);
                    clearScreen();
                    break;
                case 5:
                    System.out.println("Queue: " + songQueue);
                    clearScreen();
                    break;
                case 6:
                    songQueue.clear();
                    System.out.println("Queue cleared");
                    clearScreen();
                    break;
                case 7:
                    choice = -1;
                    System.out.print("\033[H\033[2J");
                    System.out.flush();
                    break;
                default:
                    System.out.println("Invalid choice");
            }
        }
        System.out.println("Thank you for using the Spotify Queue System");
    }

    static void clearScreen() {
        Scanner sc = new Scanner(System.in); // Declare and initialize the Scanner object
        System.out.println("Press Enter to continue");
        sc.nextLine();
        System.out.print("\033[H\033[2J");
        System.out.flush();
    }
}

